### ISSUE TITLE
[FP] SubstringCompareInspection no length

### ISSUE THREAD

**Issue by puggan** [kalessil/phpinspectionsea#1168]
| Subject        | Details                                                                       |
| :------------- | :---------------------------------------------------------------------------- |
| Issue type     | False-Positive|
| Plugin         | Php Inspections (EA Ultimate) v2.0.11 |
| Language level | PHP 7.3 |

Current behaviour (description/screenshot:)
--
SubstringCompareInspection warning, when no length is given.

```php
<?php
    $t = 'abcde';
    if(substr($t, 3) === 'de') {
        echo 'ends with de';
    }
```
![screenshot from 2019-01-16 15-37-44](https://user-images.githubusercontent.com/866668/51256190-11c60880-19a5-11e9-80c5-0bf53981ad75.png)

Expected behaviour
--
No SubstringCompareInspection warning

Environment details
---
PhpStorm 2018.3.2
Build #PS-183.4886.22, built on December 12, 2018
JRE: 1.8.0_152-release-1343-b26 amd64
JVM: OpenJDK 64-Bit Server VM by JetBrains s.r.o
Linux 4.18.0-13-generic

**Comment by puggan** [comment]
Expected:
If 3th parameter length exists, compare it with the constant string length.
else if 2th parameter is negative, compare it with the constant string length.
else skip

**Comment by kalessil** [comment]
Fixed!
